validator/main.go

```diff
@@ -675,35 +769,65 @@ func (d *Driver) validate() error {
 		}
 	}
 
-	statusFile := driverStatusFile
+	return d.createStatusFile(isHostDriver)
+}
+
+func (d *Driver) createStatusFile(isHostDriver bool) error {
+	var nvidiaDriverRoot, nvidiaDevRoot, driverRootCtrPath, devRootCtrPath string
 	if isHostDriver {
-		statusFile = hostDriverStatusFile
+		nvidiaDriverRoot = hostRootFlag
+		driverRootCtrPath = "/host"
+		nvidiaDevRoot = hostRootFlag
+		devRootCtrPath = "/host"
+	} else {
+		nvidiaDriverRoot = driverRootFlag
+		driverRootCtrPath = "/driver-root"
+		nvidiaDevRoot = devRootFlag
+
+		devRootCtrPath = driverRootCtrPath
+		if devRootFlag == hostRootFlag {
+			devRootCtrPath = "/host"
+		} else if devRootFlag != "" && devRootFlag != driverRootFlag {
+			// a custom dev root, which is not equal to host root or driver root,
+			// will be mounted in the operand container at /dev-root
+			devRootCtrPath = "/dev-root"
+		}
 	}
 
+	statusFileContent := strings.Join([]string{
+		fmt.Sprintf("NVIDIA_DRIVER_ROOT=%s", nvidiaDriverRoot),
+		fmt.Sprintf("DRIVER_ROOT_CTR_PATH=%s", driverRootCtrPath),
+		fmt.Sprintf("IS_HOST_DRIVER=%v", isHostDriver),
+		fmt.Sprintf("NVIDIA_DEV_ROOT=%s", nvidiaDevRoot),
+		fmt.Sprintf("DEV_ROOT_CTR_PATH=%s", devRootCtrPath),
```

Comment:
Nit: Can we reorder this so that `IS_HOST_DRIVER` is first?

Reply:
Done.